sympy/solvers/ode.py

```diff
@@ -679,6 +680,61 @@ def recur_len(l):
             hint = hints['hint']
             return _helper_simplify(eq, hint, hints, simplify, ics=ics)
 
+'''
+Patch here  <------------
+I would like to make it a wrapper.I have learned about decorators but haven't used them
+so I would do it if you say. 
+
+Function created to try to solve equations refered to issue
+   dsolve doesn't know order-reducing substitutions #15881
+   #Steps:
+   1: Try to use dsolve.
+
+   2: This is a recursive call till we are not left with a linear equation:
+   If NotImplemented error is raised try substituting whole expression with g(x)
+   (I think Dummy should be used but I do not understand the working of it very well 
+   so I'm letting it be a substitution function for the time being.)
+   
+   3: Still unable to solve ?Then again raise the NotImplementedError.
+   
+   #A thing I read which I would like to ask is that the given equation is a Liouville equation
+    and it's method is already implemented then why is sympy unable to solve it?
+   
+   Example :
+   from sympy.abc import x
+   from sympy import Function
+   f = Function("f")(x)
+   eq = f(x).diff(x, 2) + x * f(x).diff(x)**2
+   print(eq)
+   Eq(f(x), C1 - sqrt(-1/C2)*log(-C2*sqrt(-1/C2) + x) + sqrt(-1/C2)*log(C2*sqrt(-1/C2) + x))
+   
+'''
```

Comment:
If it could be used as a wrapper or decorator it can be good but I never used one.

Reply:
> 
> 
> Okay well done for getting this far. I know git can be hard to use!
> 
> So what we want to do know is integrate this code into the `dsolve` function so that `dsolve` can solve this type of ODE. For that we want to make a new solver within dsolve. If you read the docstring at the top of ode.py you can see that it explains about how this works. So there are three parts to adding a solver.
> 
>     1. We need to give it a name e.g. "order_reducing_substitution" and add that name to the list of methods here: https://github.com/sympy/sympy/blob/master/sympy/solvers/ode.py#L309
> 
>     2. We need to add the code to match this kind of ODE into the matching part of `dsolve` which is somewhere around here: https://github.com/sympy/sympy/blob/master/sympy/solvers/ode.py#L1362. If the ODE matches then we do `matching_hints['order_reducing_substitution'] = match`.
> 
>     3. Then we need a function called `ode_order_reducing_substitution` that will be called when this method is used. You can see an example here https://github.com/sympy/sympy/blob/master/sympy/solvers/ode.py#L4719. Whatever variables you stored as `match` will be passed through to this function to save recalculating them.

Hi @oscarbenjamin 
Will I created an inner function inside dsolve but that was giving me errors.I will implement it once again.But the problem before that is that I have crosses over my pull request ie Some checks were not successful.What to do?They were extra spacing issues which I deleted still...

Reply:
Modified it.Please tell for further amendments.

Reply:
I already gave a list of things that need doing to get this working with `dsolve`. You'll know when you have it working because `dsolve` will solve the ODE.

Reply:
I will surely do this till tomorrow since I got a test tomorrow morning.After coming back I 'll surely do it.